e2e-tests/otel-collector/provisioner.go

```diff
@@ -0,0 +1,170 @@
+	return func(params *ProvisionerParams) error {
+		params.otelOptions = append(params.otelOptions, opts...)
+		return nil
+	}
+}
+
+// WithExtraParams sets the extra parameters for the provisioner
+func WithExtraParams(params runner.ConfigMap) ProvisionerOption {
+	return func(p *ProvisionerParams) error {
+		p.extraParams = params
+		return nil
+	}
+}
+
+// Provisioner creates a new provisioner
+func Provisioner(opts ...ProvisionerOption) e2e.TypedProvisioner[Kubernetes] {
+	// We ALWAYS need to make a deep copy of `params`, as the provisioner can be called multiple times.
+	// and it's easy to forget about it, leading to hard to debug issues.
+	params := newProvisionerParams()
+	_ = optional.ApplyOptions(params, opts)
+
+	provisioner := e2e.NewTypedPulumiProvisioner(provisionerBaseID+params.name, func(ctx *pulumi.Context, env *Kubernetes) error {
+		// We ALWAYS need to make a deep copy of `params`, as the provisioner can be called multiple times.
+		// and it's easy to forget about it, leading to hard to debug issues.
+		params := newProvisionerParams()
+		_ = optional.ApplyOptions(params, opts)
+
+		return RunFunc(ctx, env, params)
+	}, params.extraParams)
+
+	return provisioner
+}
+
+// RunFunc is the Pulumi run function that runs the provisioner
+func RunFunc(ctx *pulumi.Context, env *Kubernetes, params *ProvisionerParams) error {
+	awsEnv, err := aws.NewEnvironment(ctx)
+	if err != nil {
+		return err
+	}
+
+	host, err := ec2.NewVM(awsEnv, params.name)
+	if err != nil {
+		return err
+	}
+
+	installEcrCredsHelperCmd, err := ec2.InstallECRCredentialsHelper(awsEnv, host)
+	if err != nil {
+		return err
+	}
+
+	kindCluster, err := kubeComp.NewKindCluster(&awsEnv, host, awsEnv.CommonNamer().ResourceName("kind"), params.name, awsEnv.KubernetesVersion(), utils.PulumiDependsOn(installEcrCredsHelperCmd))
+	if err != nil {
+		return err
+	}
+
+	err = kindCluster.Export(ctx, &env.KubernetesCluster.ClusterOutput)
+	if err != nil {
+		return err
+	}
+
+	kubeProvider, err := kubernetes.NewProvider(ctx, awsEnv.Namer.ResourceName("k8s-provider"), &kubernetes.ProviderArgs{
+		EnableServerSideApply: pulumi.Bool(true),
+		Kubeconfig:            kindCluster.KubeConfig,
+	})
+	if err != nil {
+		return err
+	}
+
+	fi, err := fakeintake.NewECSFargateInstance(awsEnv, params.name)
+	if err != nil {
+		return err
+	}
+	fi.Export(ctx, &env.FakeIntake.FakeintakeOutput)
+
+	otelOptions := []otelparams.Option{otelparams.WithPulumiResourceOptions(pulumi.Provider(kubeProvider)), otelparams.WithFakeintake(fi)}
+	otelOptions = append(otelOptions, params.otelOptions...)
+	otelCollector, err := helm.NewOtelCollector(&awsEnv, "otel-collector", otelOptions...)
+	if err != nil {
+		return err
+	}
+
+	otelCollector.Export(ctx, &env.OtelCollector.OtelCollectorOutput)
+
+	return nil
+}
+
+// LocalRunFunc is the Pulumi run function that runs the provisioner
+func LocalRunFunc(ctx *pulumi.Context, env *Kubernetes, params *ProvisionerParams) error {
```

Comment:
Could you provide an option that uses this to run the tests locally? I tried to use this function in `Provisioner` but had trouble pulling the collector image.

Reply:
The error is expected I think. Since the image is not a public one this time you need some authentication to be able pull the image. As you can see [here](https://github.com/DataDog/opentelemetry-collector-contrib/pull/5270/files#diff-037ea159eb0a7cb0ac23b851e66bee30fb838ee8d0d99fa331a1ba65283d37f7R165) we need to provide a `docker_secret` arg to the test run.
Locally you can do that `go test ./tests/... -v -count 1 --args -docker_secret $(aws-vault exec sso-agent-qa-read-only -- aws ecr get-login-password)`